### PR TITLE
Use protoc's --pyi_out instead of mypy-protobuf package

### PR DESCRIPTION
**Pull request by keepingitneil**
And add type checking GHA

Successful type checking run: https://github.com/livekit/python-sdks/actions/runs/8351262508

Purposely failed type checking run by adding `bad_variable: str = 5` to a few files: https://github.com/livekit/python-sdks/actions/runs/8351236763

### CODE REVIEW

File: livekit-api/livekit/api/__init__.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2023 LiveKit, Inc.
+# Copyright 2024 LiveKit, Inc.
```
Comment (davidzhao) [inline livekit-api/livekit/api/__init__.py:1]: don't need to change this

**Review by davidzhao — APPROVED** [review]

**Comment by theomonnom** [comment]
Is changing mypy to pyi generator a breaking change?

File: livekit-rtc/livekit/rtc/room.py [hunk 2]
```diff
@@ -80,9 +81,9 @@ class RoomOptions:
 class DataPacket:
     data: bytes
     kind: proto_room.DataPacketKind.ValueType
```
Comment (theomonnom) [inline livekit-rtc/livekit/rtc/room.py:83]: Weird thing is that ValueType doesn't exist anymore due to the pyi generator but mypy is not complaining ? Seems like the type checking isn't working?

**Comment by theomonnom** [comment]
The mypy generator may be a better choice for stricter type checking

File: livekit-rtc/livekit/rtc/room.py (hunk 2 above)
Comment (keepingitneil) [reply livekit-rtc/livekit/rtc/room.py:83]: Oh this is Room is not from livekit-protocol, it's from livekit/rtc/_proto which is still generated by mypy-protobuf.

Comment (keepingitneil) [reply livekit-rtc/livekit/rtc/room.py:83]: Just to double check, changed kind to something bad (i.e. `str`) and ran it again and it failed as expected: https://github.com/livekit/python-sdks/actions/runs/8362789419

**Comment by keepingitneil** [comment]
> The mypy generator may be a better choice for stricter type checking if the types are not detected (the above comment)

Yeah in my testing pyi_out has been working flawlessly. Since it's newer, a bit more "official", and  removes a build dependency I decided to make the switch. I left livekit-rtc in-tact but we can switch to pyi_out for those protos too in another PR if you're satisfied with the pyi_out typing.

**Comment by keepingitneil** [comment]
> Is changing mypy to pyi generator a breaking change?

It should not be. It only affects the pyi files. The generated py files remain as-is.

**Review by theomonnom — APPROVED** [review]

**Review by theomonnom — APPROVED** [review]
nice!

**Pull request merged**